sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -561,6 +561,39 @@ object GpuCast extends Arm {
         GpuIntervalUtils.castStringToDayTimeIntervalWithThrow(
           input.asInstanceOf[ColumnVector], dayTime)
 
+      // cast(`day time interval` as integral)
+      case (dt: DataType, _: LongType) if GpuTypeShims.isSupportedDayTimeType(dt) =>
+        GpuIntervalUtils.dayTimeIntervalToLong(input.asInstanceOf[ColumnVector], dt)
```

Comment:
But input is not guaranteed to be a `ColumnVector` for nested types. Like casting an array of DayTimeInterval to an Array of longs. These need to be ColumnView, but you should be able to treat it exactly the same as a ColumnVector.

Reply:
Done